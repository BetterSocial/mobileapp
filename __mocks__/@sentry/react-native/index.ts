export const addBreadcrumb = jest.fn();
