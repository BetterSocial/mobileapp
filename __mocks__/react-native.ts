import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN