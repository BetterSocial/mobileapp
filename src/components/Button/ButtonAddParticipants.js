import * as React from 'react';
import {useNavigation} from '@react-navigation/core';
import {StyleSheet, Text, TouchableWithoutFeedback, View} from 'react-native';
import {COLORS} from '../../utils/theme';
import {fonts} from '../../utils/fonts';

const ButtonAddParticipants = (props) => {
  const navigation = useNavigation();
  return (
    <View style={styles.btnAdd}>
      <TouchableWithoutFeedback
        onPress={() => navigation.navigate('AddParticipant', {refresh: props.refresh})}>
        <Text style={styles.btnAddText}>+ Add Participants</Text>
      </TouchableWithoutFeedback>
    </View>
  );
};

export default ButtonAddParticipants;

const styles = StyleSheet.create({
  btnAdd: {
    paddingHorizontal: 8,
    paddingVertical: 8,
    backgroundColor: COLORS.lightgrey,
    width: 'auto',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
    alignSelf: 'center'
  },
  btnAddText: {
    fontFamily: fonts.inter[600],
    fontSize: 14,
    lineHeight: 20,
    color: COLORS.holytosca
  }
});
