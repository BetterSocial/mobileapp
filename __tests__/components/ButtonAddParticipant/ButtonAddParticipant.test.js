import * as React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react-native';
import ButtonAddParticipants from '../../../src/components/Button/ButtonAddParticipants';
import {act} from 'react-test-renderer';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/core', () => ({
  ...jest.requireActual('@react-navigation/core'),
  useNavigation: () => ({
    goBack: jest.fn(),
    navigate: mockNavigate
  })
}));

describe('ButtonAddParticipant should run correctly', () => {
  it('should be match with snapshot', () => {
    const refresh = jest.fn();
    const {toJSON, getByTestId} = render(<ButtonAddParticipants refresh={refresh} />);
    act(() => {
      fireEvent.press(getByTestId('btnPress'));
    });
    expect(mockNavigate).toHaveBeenCalled();
    expect(toJSON).toMatchSnapshot();
  });
});
