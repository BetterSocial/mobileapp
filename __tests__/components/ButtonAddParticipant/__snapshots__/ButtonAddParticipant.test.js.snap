// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAddParticipant should run correctly should be match with snapshot 1`] = `[Function]`;
